include/RAJA/util/sort.hpp

```diff
@@ -548,16 +549,37 @@ merge_sort(Iter begin,
 
         if ( start + midpoint >= len )
         {
+          // copy sorted remainder over
+          if ( copyvalid )
+          {
+            std::move( copyarr + start, copyarr + finish, begin + start );
+          }
+          else
+          {
+            std::move( begin + start, begin + finish, copyarr + start );
+          }
           break;  // skip merge if no second half exists
         }
 
-        detail::merge_like_std( copyarr + start, copyarr + start + midpoint, copyarr + start + midpoint, copyarr + finish, begin + start, comp );
+        if ( copyvalid )  // switch arrays per level of merging to avoid copying back to copyarr
+        {
+          detail::merge_like_std( copyarr + start, copyarr + start + midpoint, copyarr + start + midpoint, copyarr + finish, begin + start, comp );
+        }
+        else
+        {
+          detail::merge_like_std( begin + start, begin + start + midpoint, begin + start + midpoint, begin + finish, copyarr + start, comp );
+        }
       }
 
-      // update copy
+      copyvalid = !copyvalid; // switch arrays per level of merging to avoid copying back to copyarr
+    }
+
+    // update copy if necessary
+    if ( copyvalid )
+    {
       for ( diff_type cc = 0; cc < len; ++cc )
```

Comment:
This loop looks like a good candidate for range-based [`std::move`](https://en.cppreference.com/w/cpp/algorithm/move)

Reply:
Good catch! It's now a ranged move.